src/smt/smt_driver.h

```diff
@@ -0,0 +1,106 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The solver for SMT queries in an SolverEngine.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__SMT__SMT_DRIVER_H
+#define CVC5__SMT__SMT_DRIVER_H
+
+#include <vector>
+
+#include "expr/node.h"
+#include "smt/assertions.h"
+#include "smt/env_obj.h"
+#include "util/result.h"
+
+namespace cvc5::internal {
+namespace smt {
+
+class SmtSolver;
+class ContextManager;
+
+/**
+ * SMT driver class.
+ *
+ * The purpose of this class is to define algorithms for checking
+ * satisfiability beyond a single call to the underlying SMT solver. The
+ * default implementation, SmtDriverSingleCall, is used for invoking a
+ * single call to the SMT solver only.
+ */
+class SmtDriver : protected EnvObj
+{
+ public:
+  SmtDriver(Env& env, SmtSolver& smt, ContextManager* ctx);
+
+  /**
+   * Check satisfiability. This invokes the algorithm given by this driver
+   * for checking satisfiability.
+   *
+   * @param assumptions The assumptions for this check-sat call, which are
+   * temporary assertions.
+   */
+  Result checkSatisfiability(const std::vector<Node>& assumptions);
```

Comment:
How about

```suggestion
  Result checkSat(const std::vector<Node>& assumptions);
```

to be consistent with `checkSatNext` below?